resources/lib/videostream.py

```diff
@@ -86,38 +88,44 @@ def get_player_stream_data(self) -> Optional[VideoPlayerStreamData]:
         video_player_stream_data.token = async_data.get('stream_data').get('token')
 
         video_player_stream_data.skip_events_data = async_data.get('skip_events_data')
+        video_player_stream_data.unmodified_skip_events_data = dict(async_data.get('skip_events_data'))
```

Comment:
Not sure if we really do need that, as it is pretty much the same as the line above, but with the dict wrapped in another dict (if `_get_skip_events` doesn't return `None`, that is)?

Reply:
It make a clone, to prevent for example, drop by this https://github.com/lumiru/plugin.video.crunchyroll/blob/fd0dd73d8aec9cf35564609986064ab8655a108f/resources/lib/videoplayer.py#L336

Reply:
Hm I see. If that's the only reason, you could move `self._handle_upnext()` in 65 up before `self._handle_skipping()` in ` videoplayer.py` 

```python
        self._handle_update_playhead()
        self._handle_upnext()  # <-
        self._handle_skipping()

        if not self._wait_for_playback_started(10):
            utils.crunchy_log('Timeout reached, video did not start playback in 10 seconds', xbmc.LOGERROR)

        # self._handle_upnext()
```